eras/conway/impl/src/Cardano/Ledger/Conway/TxInfo.hs

```diff
@@ -450,7 +451,7 @@ instance Crypto c => EraPlutusTxInfo 'PlutusV3 (ConwayEra c) where
         , PV3.txInfoOutputs = outputs
         , PV3.txInfoReferenceInputs = refInputs
         , PV3.txInfoFee = transCoinToLovelace (txBody ^. feeTxBodyL)
-        , PV3.txInfoMint = Alonzo.transMultiAsset (txBody ^. mintTxBodyL)
+        , PV3.txInfoMint = PV3.UnsafeMintValue . PV1.getValue $ Alonzo.transMultiAsset (txBody ^. mintTxBodyL)
```

Comment:
I don't like having to go outside the standard Plutus API and use a private constructor. However, this is what the Plutus team have requested.

I also don't like that we now have an existing function `transMintValue` that returns a `Value` and not a `MintValue`. I'm wondering whether it should be simply be an alias for `transMultiAsset` rather than having its own duplicate implementation, or whether we should remove it entirely and replace it with direct calls to `transMultiAsset`.

Reply:
> Difference between MintValue and MultiAsset is that former can contain negative values, while latter can't. So they are not synonymous.

Agreed, but that means it should be safe to convert `MultiAsset` to `MintValue`, even though the other direction doesn't work.

> The only suggestion I have is to extract this into a separate function, since I suspect it will be also used for PlutusV4 in the future:
> 
> ```haskell
> transMintValue = PV3.UnsafeMintValue . PV1.getValue . Alonzo.transMultiAsset
> ```

I was wondering about that, but we already have a `transMintValue` in Alonzo. It's different, though, because it returns a `V1.Value` rather than a `V3.MintValue`, which is a bit confusing. Is it OK to have another version with the same name in Conway, or should we make some distinction in the names, such as `transV3MintValue`?

Alternatively, since the existing `transMintValue` is identical to `transMultiAsset`: 

```haskell
transMintValue :: MultiAsset c -> PV1.Value
transMintValue m = transMultiAssetInternal m (transCoinToValue zero)

transMultiAsset :: MultiAsset c -> PV1.Value
transMultiAsset ma = transMultiAssetInternal ma mempty
```

we could remove the current `transMintValue` and use `transMultiAsset` instead. We can then use the new `transMintValue` you've given above for the V3 situation. I think that would be a lot less confusing, and it seems to me that the semantic hint provided by the name of the old `transMintValue` isn't useful since in every place where it's used the context shows that the `V1.Value` is being used as a mint value.

Reply:
For now, I've added another `transMintValue` function in Conway and I haven't touched the other one.

Reply:
> I've added another transMintValue function in Conway and I haven't touched the other one.

That is the correct way to go about this. We do not want to make any changes to packages that implement unaffected plutus versions (i.e. no changes should be needed to Alonzo and Babbage eras)

>  It's different, though, because it returns a V1.Value rather than a V3.MintValue, which is a bit confusing.

PlutusV1 and V2 still use `Value`, not `MintValue`, so that is expected.

> should we make some distinction in the names, such as transV3MintValue?

That is not the convention we adopted. We are expected to use same names whenever types change. That is why you'll find qualified imports from previous eras (eg. `Alonzo.transMultiAsset`)

> we could remove the current transMintValue and use transMultiAsset instead

They have different semantics, so `transMultiAsset` can't be used instead of `transMintValue`.

Good thing you didn't change anything in previous eras. We should always be extremely careful when we are changing anything in plutus context for active eras, it is very easy to introduce dangerous bugs that change behavior of existing scripts.